pkg/provisioningv2/kubeconfig/manager.go

```diff
@@ -131,13 +132,19 @@ func getPrincipalID(clusterNamespace, clusterName string) string {
 	return fmt.Sprintf("system://provisioning/%s/%s", clusterNamespace, clusterName)
 }
 
-func (m *Manager) createUser(principalID, userName string) error {
+func (m *Manager) createUser(principalID, userName, clusterName string, clusterUID types.UID) error {
 	_, err := m.userCache.Get(userName)
 	if apierror.IsNotFound(err) {
 		_, err = m.users.Create(&v3.User{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:   userName,
 				Labels: labelsForUser(principalID),
+				OwnerReferences: []metav1.OwnerReference{{
+					APIVersion: "provisioning.cattle.io/v1",
+					Kind:       "Cluster",
+					Name:       clusterName,
```

Comment:
I don't think this will work. User is cluster-scoped resource and cluster.provisioning.cattle.io is namespaced scoped. https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/ Cluster-scoped resource can only have cluster-scoped resource as owner.

Reply:
Perhaps, I should just delete it then in the "mutually assured destruction" cluster removal code then?